src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp

```diff
@@ -75,26 +75,53 @@ QVariant QgsValueRelationWidgetWrapper::value() const
       }
     }
 
-    QVariantList vl;
-    //store as QVariantList because it's json
-    for ( const QString &s : qgis::as_const( selection ) )
+    if ( layer()->fields().at( fieldIdx() ).type() == QVariant::Map )
```

Comment:
Does this need to check for list too?

Reply:
Not sure by heart. I thought it has always the type map. But I will doublecheck.

Reply:
Well apparently the content  is handled as JSON (means `QVariant::Map` or as this hstore-like list-format `{1,2,3}}` (means: `QVariant::String`) so we do not need to check for list. Or am I'm wrong?

Reply:
```
git grep -n QVariant::List
arcgisrest/qgsarcgisrestutils.cpp:152:    if ( curveData.type() == QVariant::List )
oracle/ocispatial/qsql_ocispatial.cpp:1850:    fieldTypes.append( tp == QVariant::List ? boundValues.at( i ).toList().value( 0 ).type()
oracle/ocispatial/qsql_ocispatial.cpp:1863:    if ( boundValues.at( i ).type() != QVariant::List )
oracle/ocispatial/qsql_ocispatial.cpp:2138:    if ( tp != QVariant::List )
postgres/qgspostgresconn.cpp:1053:    case QVariant::List:
postgres/qgspostgresprovider.cpp:238:      << QgsVectorDataProvider::NativeType( tr( "Array of number (integer - 32bit)" ), QStringLiteral( "int4[]" ), QVariant::List, -1, -1, -1, -1, QVariant::Int )
postgres/qgspostgresprovider.cpp:239:      << QgsVectorDataProvider::NativeType( tr( "Array of number (integer - 64bit)" ), QStringLiteral( "int8[]" ), QVariant::List, -1, -1, -1, -1, QVariant::LongLong )
postgres/qgspostgresprovider.cpp:240:      << QgsVectorDataProvider::NativeType( tr( "Array of number (double)" ), QStringLiteral( "double precision[]" ), QVariant::List, -1, -1, -1, -1, QVariant::Double )
postgres/qgspostgresprovider.cpp:1053:        fieldType = ( fieldType == QVariant::String ? QVariant::StringList : QVariant::List );
postgres/qgspostgresprovider.cpp:3838:    case QVariant::List:
postgres/qgspostgresprovider.cpp:4459:  if ( ( type == QVariant::StringList || type == QVariant::List ) && inner.startsWith( "{" ) )
postgres/qgspostgresprovider.cpp:4479:    case QVariant::List:
spatialite/qgsspatialitefeatureiterator.cpp:580:    if ( type == QVariant::List || type == QVariant::StringList )
spatialite/qgsspatialiteprovider.cpp:95:    case QVariant::List:
spatialite/qgsspatialiteprovider.cpp:605:                  << QgsVectorDataProvider::NativeType( tr( "Array of decimal numbers (double)" ), SPATIALITE_ARRAY_PREFIX.toUpper() + "REAL" + SPATIALITE_ARRAY_SUFFIX.toUpper(), QVariant::List, 0, 0, 0, 0, QVariant::Double )
spatialite/qgsspatialiteprovider.cpp:606:                  << QgsVectorDataProvider::NativeType( tr( "Array of whole numbers (integer)" ), SPATIALITE_ARRAY_PREFIX.toUpper() + "INTEGER" + SPATIALITE_ARRAY_SUFFIX.toUpper(), QVariant::List, 0, 0, 0, 0, QVariant::LongLong )
spatialite/qgsspatialiteprovider.cpp:664:    return TypeSubType( subType.first == QVariant::String ? QVariant::StringList : QVariant::List, subType.first );
spatialite/qgsspatialiteprovider.cpp:668:    return TypeSubType( QVariant::List, QVariant::Invalid );
spatialite/qgsspatialiteprovider.cpp:4097:          else if ( type == QVariant::StringList || type == QVariant::List )
spatialite/qgsspatialiteprovider.cpp:4404:        else if ( type == QVariant::StringList || type == QVariant::List )
```

I guess the one where spatialite provides arrays as lists could potentially be interesting for us here?

Reply:
Hm... okay, havent seen that. We can consider that and add the check for list.